rust_bot/crates/client/src/key_words.rs

```diff
@@ -76,6 +76,23 @@ impl KeyWordManager<'_> {
         })
     }
 
+    pub async fn update(&self, key_word: &KeyWord) -> ThiccResult<KeyWord> {
+        let res = self.client.put_json(
+            format!("{}/{}", &self.guild_route, key_word.name),
+            key_word
+        ).await;
+        ThiccClient::handle_status(res, |status| {
+            if status == reqwest::StatusCode::BAD_REQUEST {
+                Some(ThiccError::ResourceAlreadyExist {
+                    name: key_word.name.clone(),
+                    resource_type: "Key Word".to_string(),
+                })
```

Comment:
O wait actually does this happen? I think it would be a 404 instead of a 400, can you add a `ResourceDoesNotExist` error here https://github.com/JRMurr/ThiccBot/blob/master/rust_bot/crates/client/src/error.rs#L29

Reply:
Yeah my bad, I updated this to catch the 404 and throw the new error.